src/components/AddCardForm/AddCardForm.tsx

```diff
@@ -0,0 +1,88 @@
+import React from 'react';
+import styles from './AddCardForm.module.css';
+
+import { INITIAL_VALUES } from '../../constants/form';
+
+import CVCInput from '../AddCardFormInput/CVCInput/CVCInput';
+import CardIssuerInput from '../AddCardFormInput/CardIssuerInput/CardIssuerInput';
+import CardNumberInput from '../AddCardFormInput/CardNumberInput/CardNumberInput';
+import ExpirationDateInput from '../AddCardFormInput/ExpirationDateInput/ExpirationDateInput';
+import OwnerNameInput from '../AddCardFormInput/OwnerNameInput/OwnerNameInput';
+import PasswordInput from '../AddCardFormInput/PasswordInput/PasswordInput';
+import CardPreview from '../CardPreview/CardPreview';
+import Button from '../common/Button/Button';
+
+import { useNavigate } from 'react-router-dom';
+import {
+  AddCardFormContextType,
+  useAddCardFormContext,
+} from '../../context/AddCardFormContext';
+import useAddCardFormField from '../../hooks/useAddCardFormField';
+
+export default function AddCardForm() {
+  const { findStep, curStep, isFormValid } =
+    useAddCardFormContext() as AddCardFormContextType;
+
+  const cardNumbersProps = useAddCardFormField<CardNumbers>({
+    initialValues: INITIAL_VALUES.cardNumbers,
+  });
+  const expirationDateProps = useAddCardFormField<ExpirationDate>({
+    initialValues: INITIAL_VALUES.expirationDate,
+  });
+  const ownerNameProps = useAddCardFormField<OwnerName>({
+    initialValues: INITIAL_VALUES.ownerName,
+  });
+  const cardIssuerProps = useAddCardFormField<CardIssuer>({
+    initialValues: INITIAL_VALUES.cardIssuer,
+  });
+  const cvcProps = useAddCardFormField<CVC>({
+    initialValues: INITIAL_VALUES.cvc,
+  });
+  const passwordProps = useAddCardFormField<Password>({
+    initialValues: INITIAL_VALUES.password,
+  });
```

Comment:
### 🤔 질문 1
[[step 1 관련 코멘트]](https://github.com/woowacourse/react-payments/pull/360#discussion_r1571576522)

step1에서 App.tsx와 useAddCardInput 훅에서 중복으로 관리되던 데이터 구조를 개선하기 위해 아래와 같은 부분을 고쳤습니다.
- `useAddCardInput` -> `useAddCardFormField` 네이밍 변경.
- 부모 컴포넌트(AppCardForm)에서 `useAddCardFormField()`를 호출해 필드에서 필요한 state와 함수들을 각 컴포넌트로 전달.

그런데 `AppCardForm`만 봐서는 어떤 prop들이 전달되는지 모르기도 하고, input의 값들이 form의 상태와는 따로 관리되고 있어서 데이터 흐름을 파악하기 쉽지 않아진 것 같습니다.😅

하루는 이 코드에 대해서 어떻게 생각하시나요?

Reply:
> AppCardForm만 봐서는 어떤 prop들이 전달되는지 모르기도 하고

오 저는, `INITIAL_VALUES`를 선언해서 관리하고 계신 constant 파일을 보면서, 빈 문자열 밖에 없어서 (명세 역할은 하지만) 상수로 관리할 필요가 있을까? 하는 생각이 들었었거든요. 이걸 그냥 AddCardForm 내부로 들고 오면 어떨까요? 그러면 어떤 구조인지 잘 보일 것 같다는 생각에 제안드려보아요.

> input의 값들이 form의 상태와는 따로 관리되고 있어서

저는 두 레이어가 나뉜 것은 좋아보여요! 
폼 전체를 총괄하는 curStep, isFormValid 같은 상태는 context로 구성해 프로바이더로 내려주고, 
각 input값을 관리하는 쪽에서 필요시 참조하거나 업데이트하는 지금의 방식 좋아보입니다.

혹시 Provider 대신 모두 props로 전달하면 데이터의 흐름을 직관적으로 이해하는 데 도움이 될 것 같다고 생각되신다면 (지금 prop driling이 심한 상황은 아니니) 모두 props로 내려주는 방법도 있을 것 같아요 

e.g. `<<CardNumberInput {...cardNumbersProps} {...formCommonProps} />` (현재도 좋습니다.)

Reply:
#### 1. initialValues를 AddCardForm 내부로 들고 오기
initialValues들의 구조가 객체라서 코드가 너무 길어질까봐 상수로 분리했던 것이었는데, 오히려 input들의 구조가 잘 안 보이게 됐네요..! 한번 고민해 보겠습니다.🫡

#### 2. form 상태와 input 상태 레이어 분리
하루는 레이어로 분리된 현재 상태가 어색하지 않다고 느끼는군요..! 다행입니다. 
props로 내려주는 방법도 구현할 때 고려했지만 인풋 필드에서 props로 한꺼번에 받으면 두 레이어가 분리되어 있다고 생각하지 못할 것 같아서 context로 구성하게 되었습니다.

Reply:
> props로 한꺼번에 받으면 두 레이어가 분리되어 있다고 생각하지 못할 것 같아서 context로 구성하게 되었습니다.

아하, Context로 구성해주셨던 이유가 그것이었군요. 

그런데 Depth가 동일한데, 계층을 분리하기 위해서 context를 사용해주신 것은 제게는 조금 생소합니다. 리액트 공식문서에서도 props 부터 사용해볼 것을 권장하고 있습니다. `formProps` 등 props의 네이밍을 통해 레이어를 구분하는 방법은 어떨까요?

> Start by passing props. If your components are not trivial, it’s not unusual to pass a dozen props down through a dozen components. It may feel like a slog, but it makes it very clear which components use which data! The person maintaining your code will be glad you’ve made the data flow explicit with props.

- 참고: [React 공식 문서 - Passing Data Deeply with Context](https://react.dev/learn/passing-data-deeply-with-context#before-you-use-context)